File: src/main/scala/com/exasol/cloudetl/bucket/HDFSBucket.scala
package com.exasol.cloudetl.bucket

import com.exasol.cloudetl.storage.StorageProperties

import org.apache.hadoop.conf.Configuration

/** A [[Bucket]] implementation for the HDFS filesystem */
final case class HDFSBucket(path: String, params: StorageProperties) extends Bucket {

  /** @inheritdoc */
  override val bucketPath: String = path

  /** @inheritdoc */
  override val properties: StorageProperties = params

  /** Returns the list of required property keys for HDFS filesystem. */
  override def getRequiredProperties(): Seq[String] = Seq.empty[String]

  /** @inheritdoc */
  override def validate(): Unit =
    validateRequiredProperties()

  /** @inheritdoc */
  override def getConfiguration(): Configuration = {
    validate()
    val conf = new Configuration()
    conf.set("fs.hdfs.impl", classOf[org.apache.hadoop.hdfs.DistributedFileSystem].getName())
    conf
  }

}

File: src/main/java/com/exasol/cloudetl/bucket/HDFSBucket.java
package com.exasol.cloudetl.bucket;

import java.util.List;

import org.apache.hadoop.conf.Configuration;

import com.exasol.cloudetl.ScalaConverters;
import com.exasol.cloudetl.storage.StorageProperties;

/** Bucket implementation for HDFS. */
public final class HDFSBucket extends AbstractConfiguredBucket {
    /** Create a bucket. */
    public HDFSBucket(final String path, final StorageProperties params) {
        super(path, params);
    }

    @Override
    public boolean equals(final Object obj) {
        return (this == obj) || ((obj instanceof HDFSBucket) && hasSameConfiguration((HDFSBucket) obj));
    }

    @Override
    public int hashCode() {
        return configuredHashCode(HDFSBucket.class);
    }

    @Override
    public scala.collection.immutable.Seq<String> getRequiredProperties() {
        return ScalaConverters.seqFromJava(List.of());
    }

    @Override
    public void validate() {
        validateRequiredProperties();
    }

    @Override
    public Configuration getConfiguration() {
        validate();
        final Configuration conf = new Configuration();
        conf.set("fs.hdfs.impl", org.apache.hadoop.hdfs.DistributedFileSystem.class.getName());
        return conf;
    }
}
